githubwebhooks/iam-roles.tf

```diff
@@ -146,3 +146,51 @@ resource "aws_iam_role_policy" "lambda_github_webhooks_pulse" {
     role = "${aws_iam_role.lambda_github_webhooks_pulse.id}"
     policy = "${data.aws_iam_policy_document.lambda_github_webhooks_pulse.json}"
 }
+
+data "aws_iam_policy_document" "sns_webhooks_all" {
+    # Give self full access.
+    statement = {
+        sid = "github_webhooks_all_self_access"
+        effect = "Allow"
+        actions = [
+            "SNS:AddPermission",
+            "SNS:DeleteTopic",
+            "SNS:GetTopicAttributes",
+            "SNS:ListSubscriptionsByTopic",
+            "SNS:Publish",
+            "SNS:RemovePermission",
+            "SNS:SetTopicAttributes",
+            "SNS:Subscribe"
+        ]
+        resources = [
+            "${aws_sns_topic.webhooks_all.arn}",
+        ]
+        principals {
+            type = "AWS"
+            identifiers = ["*"],
+        }
+        condition {
+            test = "StringEquals"
+            variable = "AWS:SourceOwner",
+            values = [
+                "${data.aws_caller_identity.current.account_id}",
+            ]
+        }
+    }
+
+    # Grant subscribe access to infosec-prod account.
+    statement = {
+        sid = "github_webhooks_all_infosec_subscribe"
+        effect = "Allow"
+        actions = [
+            "SNS:Subscribe",
```

Comment:
Could you also grant [ListSubscriptions](http://docs.aws.amazon.com/sns/latest/api/API_ListSubscriptions.html) and [Unsubscribe](http://docs.aws.amazon.com/sns/latest/api/API_Unsubscribe.html) so we can determine our subscription ARN and unsubscribe if needed?